frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -995,6 +1098,53 @@ void Visitor::visit(const Import* node) {
   }
 }
 
+template <typename F, typename NodeType>
```

Comment:
Add a comment describing what this function does? Mention it will call the passed function on each parent node?